src/orthonormal.jl

```diff
@@ -62,6 +62,19 @@ LinearAlgebra.mul!(y, b::OrthonormalBasis, x::AbstractVector) = unproject!!(y, b
 
 const BLOCKSIZE = 4096
 
+# helper function to determine if a multithreaded approach should be used
+# this uses functionality beyond VectorInterface, but can be faster
+_use_multithreaded_array_kernel(y) = false
```

Comment:
Maybe, in order to avoid duplication of cases:
```julia
_use_multithreaded_array_kernel(y) = _use_multithreaded_array_kernel(typeof(y))
_use_multithreaded_array_kernel(::Type) = false
function _use_multithreaded_array_kernel(::Type{<:Array{T}}) where {T<:Number}
     return isbitstype(T) && get_num_threads() > 1
 end
 function _use_multithreaded_array_kernel(::Type{<:OrthonormalBasis{T}}) where {T}
     return _use_multithreaded_array_kernel(T)
 end
```